### ISSUE TITLE
SEF URL in Action URL problem

### ISSUE THREAD

**Issue by meangczac** [PHPFusion/PHPFusion#475]
First, all our current forms are attached with FUSION_ROOT in seo mode, escaping permalink translations. I'm testing on comments form at this moment and when i post from:

```
http://..../blogs/1/blog-title
```

it post to

```
http://.../blogs/1/blogs/1/blog-title
```

https://wordpress.org/support/topic/permalink-problems-22.

Although router is reading fine, and the path is definitely wrong. 

FUSION_ROOT was my crude hack since beta 1 to get Ankur's SEO Engine working on the forms.I was waiting for developers to fix/contribute but we are at beta 7 now, and nobody did anything yet. 

Did you know that when SEO is turned on, all blog.php?cat_id=1 is considered void by a 301 declaration? It tells all search engine that the page is permanently moved. And using the non-friendly url in form actions clearly contradicts this. 

The form is not posting at Build 2.03, clearly because it's working and everything is redirected (So that is correct). But now the problem is this, that's why redirect kicks in the first place.

I want to know why and where went wrong.

**Comment by meangczac** [comment]
POST /php-fusion/blogs/1/blogs/1/123456789 HTTP/1.1
Host: localhost
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,_/_;q=0.8
Accept-Encoding: gzip, deflate
Accept-Language: en-US,en;q=0.8
Content-Type: application/x-www-form-urlencoded
Cookie: fusionZIs8Y_lastvisit=1447805376; fusionZIs8Y_user=1.1447981778.17fe9e05fe5ee9a92dde38b416639abf412367f558c93b27d999e3277994c49b; fusionZIs8Y_session=blo5ak9keeegbbdoitj1rrif23; fusionZIs8Y_visited=yes
Origin: http://localhost
Referer: http://localhost/php-fusion/blogs/1/123456789
Upgrade-Insecure-Requests: 1
User-Agent: Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/46.0.2490.86 Safari/537.36

HTTP/1.1 301 Moved Permanently
Connection: Keep-Alive
Content-Encoding: gzip
Content-Length: 20
Content-Type: text/html; charset=UTF-8
Date: Wed, 18 Nov 2015 01:29:16 GMT
Keep-Alive: timeout=5, max=100
Location: http://localhost/php-fusion/blogs/1/123456789
Server: Apache/2.2.27 (Win32) DAV/2 mod_ssl/2.2.27 OpenSSL/1.0.1g mod_fcgid/2.3.9 mod_wsgi/3.4 Python/2.7.6 PHP/5.6.8 mod_perl/2.0.8 Perl/v5.16.3
Vary: Accept-Encoding
X-Powered-By: PHP/5.6.8

**Comment by meangczac** [comment]
I just cloned Ankur's finalRewrite copy to double confirm. Same

$target_url = string(14) "1/article/asda" 
$request_uri = string(24) "1/article/1/article/asda"

I turned on articles permalink and used the comment form in article full view.

I turned off all mpredirect().

Comment Form HTML:

```
<form action url = "1/article/asda" method="post">
...
</form>
```

Hit Post. Returned 404

<strong>The Post is Sending to Wrong URL</strong>

This is the standard stock debugger after I click Post Comment button (mpredirect off)

Path to File: 
Request URI: 1/article/1/article/asda
1. 1/article/1/article/asda : No matching Alias found.
2. 1/article/1/article/asda : No matching Alias Pattern found.
3. 1/article/1/article/asda : No matching Regex pattern found.
4. File path is empty.

What I did with mpredirect() debugging: With all the mpRedirect before invoke, a condition must be detected. If the $target_url and the $current_uri does not match, a mpredirect sending a 301 permanently moved and a redirect happens. <strong>You lose all your POST Data if redirected during a post</strong>

```
    // Now check if the CURRENT URI matches with actual URL, which it should be
                                            if (strcmp($target_url, $current_uri) != 0) {
                                                var_dump($target_url);
                                                var_dump($current_uri);


                                                //$this->mpRedirect($target_url);
                                            }
```

See the logic is there, but cannot be executed.

We are clearly missing something.

Why is the $target_url appending a 1/article/ in front on POST.
